File: Assets/Scripts/PlayerBehavior.cs
using UnityEngine;
using System.Collections;

public class PlayerBehavior : MonoBehaviour {

	public int jumpHeight;
	public int moveSpeed;
	public int maxJumps; // maximum number of jumps
	
	private int numJumps; // number of jumps

	private bool facingRight; // is our character facing right?

	// Use this for initialization
	void Start () {
		numJumps = 0;
		facingRight = true;
	}
	
	// Update is called once per frame
	void Update () {
		if (Input.GetKeyDown(KeyCode.Space) && CanJump ())
		{
			float x = GetComponent<Rigidbody2D>().velocity.x;

			GetComponent<Rigidbody2D>().velocity = new Vector2(x, jumpHeight);

			++numJumps;
		}

		if (Input.GetKey (KeyCode.LeftArrow)) 
		{
			float y = GetComponent<Rigidbody2D>().velocity.y;

			GetComponent<Rigidbody2D>().velocity = new Vector2(-moveSpeed, y);

			// if facing to the right, flip to face left
			if (facingRight) {
				Flip ();
			}
		}

		if (Input.GetKey (KeyCode.RightArrow)) 
		{
			float y = GetComponent<Rigidbody2D>().velocity.y;
			
			GetComponent<Rigidbody2D>().velocity = new Vector2(moveSpeed, y);

			// if not facing to the right, flip to face right
			if (!facingRight) {
				Flip ();
			}
		}
	}

	void OnCollisionEnter2D (Collision2D coll) 
	{
		if (coll.gameObject.CompareTag("Ground")) 
		{
			numJumps = 0;
		}
	}

	bool CanJump()
	{
		return numJumps < maxJumps;
	}

	// flips the sprite to face the opposite direction on the x-axis
	void Flip()
	{
		Vector3 flipScale; 

		Rigidbody2D rigidBody = GetComponent<Rigidbody2D>();

		flipScale = rigidBody.transform.localScale; // current scale vector
		flipScale.x *= -1; // flip horizontally, along x-axis

		rigidBody.transform.localScale = flipScale;

		facingRight = !facingRight; // facing opposite direction, now
	}
}

File: Assets/Scripts/PlayerBehavior.js
#pragma strict

public var jumpHeight = 0;
public var moveSpeed = 0;
public var maxJumps = 0; // maximum number of jumps

private var numJumps = 0; // number of current jumps
private var facingRight = true; // initially sprite faces to the right

function Start () {

}

function Update () {
	var x;
	var y;
	
	if (Input.GetKeyDown(KeyCode.Space) && CanJump()) {
		x = GetComponent(Rigidbody2D).velocity.x;
		
		GetComponent(Rigidbody2D).velocity = new Vector2(x, jumpHeight);
		
		++numJumps;
	}
	
	if (Input.GetKey (KeyCode.LeftArrow)) {
		y = GetComponent(Rigidbody2D).velocity.y;
		
		GetComponent(Rigidbody2D).velocity = new Vector2(-moveSpeed, y);
		
		if (facingRight) {
			Flip();
		}
	}
	
	if (Input.GetKey (KeyCode.RightArrow)) {
		y = GetComponent(Rigidbody2D).velocity.y;
		
		GetComponent(Rigidbody2D).velocity = new Vector2(moveSpeed, y);
		
		if (!facingRight) {
			Flip();
		}
	}
}

function OnCollisionEnter2D (coll : Collision2D) {
	if (coll.gameObject.CompareTag("Ground")) {
		numJumps = 0;
	}
}

function CanJump() {
	return numJumps < maxJumps;
}

// flip sprite along x-axis
function Flip() {
	var flipScale : Vector3;
	var rigidbody : Rigidbody2D;
	
	rigidbody = GetComponent(Rigidbody2D);
	
	flipScale = rigidbody.transform.localScale;
	flipScale.x *= -1; // flip horizontally
	
	rigidbody.transform.localScale = flipScale;
	
	facingRight = !facingRight; // facing opposite direction
}